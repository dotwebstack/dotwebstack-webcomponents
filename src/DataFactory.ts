import Quad from './Quad';
import { BlankNode, DefaultGraph, Literal, NamedNode, Variable } from './term';

class DataFactory {

  namedNode(value: string): NamedNode {
    return new NamedNode(value);
  }

  blankNode(value: string): BlankNode {
    return new BlankNode(value);
  }

  literal(value: string, languageOrDatatype?: string | NamedNode): Literal {
    if (languageOrDatatype instanceof NamedNode) {
      return new Literal(value, undefined, languageOrDatatype);
    }

    if (languageOrDatatype !== undefined) {
      return new Literal(value, languageOrDatatype);
    }

    return new Literal(value);
  }

  variable(value: string): Variable {
    return new Variable(value);
  }

  defaultGraph(): DefaultGraph {
    return new DefaultGraph();
  }

  triple(
      subject: BlankNode | NamedNode | Variable,
      predicate: NamedNode | Variable,
      object: BlankNode | NamedNode | Literal | Variable) {
    return new Quad(subject, predicate, object);
  }

  quad(
      subject: BlankNode | NamedNode | Variable,
      predicate: NamedNode | Variable,
      object: BlankNode | NamedNode | Literal | Variable,
      graph?: BlankNode | DefaultGraph | NamedNode | Variable) {
    return new Quad(subject, predicate, object, graph);
  }

}

export default DataFactory;
