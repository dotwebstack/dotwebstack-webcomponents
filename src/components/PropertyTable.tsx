import * as React from 'react';
import { connect } from 'react-redux';
import GraphState from '../graph/GraphState';
import DataFactory from '../DataFactory';
import Quad from '../Quad';

const dataFactory = new DataFactory();

export interface StateProps {
  propertyQuads: Quad[];
}

export interface OwnProps {
  resource: string;
}

export interface Props extends StateProps, OwnProps {}

const PropertyTable: React.StatelessComponent<Props> = ({ resource, propertyQuads }) => (
  <section>
    <h3>Resource: {resource}</h3>
    <table className="table">
      <thead>
        <tr>
          <th>Predicate</th>
          <th>Object</th>
        </tr>
      </thead>
      <tbody>
        {propertyQuads.map(quad => (
          <tr key={quad.predicate.value.concat(quad.object.value)}>
            <td>{quad.predicate.value}</td>
            <td>{quad.object.value}</td>
          </tr>
        ))}
      </tbody>
    </table>
  </section>
);

const mapStateToProps = (state: GraphState, ownProps: OwnProps): StateProps => ({
  propertyQuads: state.quads.filter(quad => quad.subject.equals(
    dataFactory.namedNode(ownProps.resource),
  )),
});

export default connect<StateProps>(mapStateToProps)(PropertyTable);
