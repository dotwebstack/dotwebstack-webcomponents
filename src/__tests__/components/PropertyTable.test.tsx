import React from 'react';
import { shallow } from 'enzyme';
import { Table } from 'reactstrap';
import DataFactory from '../../DataFactory';
import PropertyTable from '../../components/PropertyTable';

const dataFactory = new DataFactory();

describe('PropertyTable::render', () => {
  it('should pass table props when given', () => {
    const wrapper = shallow(<PropertyTable quads={[]} tableProps={{ striped: true }} />);
    expect(wrapper.find(Table).prop('striped')).toEqual(true);
  });

  it('should render predicate-literal pairs', () => {
    const wrapper = shallow(<PropertyTable quads={[
      dataFactory.quad(
        dataFactory.namedNode('foo'),
        dataFactory.namedNode('bar'),
        dataFactory.literal('baz'),
      ),
    ]} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render predicate-node pairs', () => {
    const wrapper = shallow(<PropertyTable quads={[
      dataFactory.quad(
        dataFactory.namedNode('foo'),
        dataFactory.namedNode('bar'),
        dataFactory.namedNode('baz'),
      ),
    ]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
