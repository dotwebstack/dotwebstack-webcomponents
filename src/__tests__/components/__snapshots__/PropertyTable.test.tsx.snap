// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyTable::render should render predicate-literal pairs 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PropertyTable
    quads={
      Array [
        Quad {
          "graph": DefaultGraph {
            "termType": "DefaultGraph",
            "value": "",
          },
          "object": Literal {
            "dataType": NamedNode {
              "termType": "NamedNode",
              "value": "http://www.w3.org/2001/XMLSchema#string",
            },
            "language": "",
            "termType": "Literal",
            "value": "baz",
          },
          "predicate": NamedNode {
            "termType": "NamedNode",
            "value": "bar",
          },
          "subject": NamedNode {
            "termType": "NamedNode",
            "value": "foo",
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Table
        responsiveTag="div"
        tag="table"
      >
        <tbody>
          <tr>
            <th
              scope="row"
            >
              <a
                href="bar"
              >
                <Connect(Component)
                  graph={
                    DefaultGraph {
                      "termType": "DefaultGraph",
                      "value": "",
                    }
                  }
                  resource={
                    NamedNode {
                      "termType": "NamedNode",
                      "value": "bar",
                    }
                  }
                />
              </a>
            </th>
            <td>
              baz
            </td>
          </tr>
        </tbody>
      </Table>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <tbody>
          <tr>
            <th
              scope="row"
            >
              <a
                href="bar"
              >
                <Connect(Component)
                  graph={
                    DefaultGraph {
                      "termType": "DefaultGraph",
                      "value": "",
                    }
                  }
                  resource={
                    NamedNode {
                      "termType": "NamedNode",
                      "value": "bar",
                    }
                  }
                />
              </a>
            </th>
            <td>
              baz
            </td>
          </tr>
        </tbody>,
        "responsiveTag": "div",
        "tag": "table",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <tr>
              <th
                scope="row"
              >
                <a
                  href="bar"
                >
                  <Connect(Component)
                    graph={
                      DefaultGraph {
                        "termType": "DefaultGraph",
                        "value": "",
                      }
                    }
                    resource={
                      NamedNode {
                        "termType": "NamedNode",
                        "value": "bar",
                      }
                    }
                  />
                </a>
              </th>
              <td>
                baz
              </td>
            </tr>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "barbaz",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th
                  scope="row"
                >
                  <a
                    href="bar"
                  >
                    <Connect(Component)
                      graph={
                        DefaultGraph {
                          "termType": "DefaultGraph",
                          "value": "",
                        }
                      }
                      resource={
                        NamedNode {
                          "termType": "NamedNode",
                          "value": "bar",
                        }
                      }
                    />
                  </a>
                </th>,
                <td>
                  baz
                </td>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    href="bar"
                  >
                    <Connect(Component)
                      graph={
                        DefaultGraph {
                          "termType": "DefaultGraph",
                          "value": "",
                        }
                      }
                      resource={
                        NamedNode {
                          "termType": "NamedNode",
                          "value": "bar",
                        }
                      }
                    />
                  </a>,
                  "scope": "row",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Connect(Component)
                      graph={
                        DefaultGraph {
                          "termType": "DefaultGraph",
                          "value": "",
                        }
                      }
                      resource={
                        NamedNode {
                          "termType": "NamedNode",
                          "value": "bar",
                        }
                      }
                    />,
                    "href": "bar",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "graph": DefaultGraph {
                        "termType": "DefaultGraph",
                        "value": "",
                      },
                      "resource": NamedNode {
                        "termType": "NamedNode",
                        "value": "bar",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "a",
                },
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "baz",
                },
                "ref": null,
                "rendered": "baz",
                "type": "td",
              },
            ],
            "type": "tr",
          },
        ],
        "type": "tbody",
      },
      "type": [Function],
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Table
          responsiveTag="div"
          tag="table"
        >
          <tbody>
            <tr>
              <th
                scope="row"
              >
                <a
                  href="bar"
                >
                  <Connect(Component)
                    graph={
                      DefaultGraph {
                        "termType": "DefaultGraph",
                        "value": "",
                      }
                    }
                    resource={
                      NamedNode {
                        "termType": "NamedNode",
                        "value": "bar",
                      }
                    }
                  />
                </a>
              </th>
              <td>
                baz
              </td>
            </tr>
          </tbody>
        </Table>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <tbody>
            <tr>
              <th
                scope="row"
              >
                <a
                  href="bar"
                >
                  <Connect(Component)
                    graph={
                      DefaultGraph {
                        "termType": "DefaultGraph",
                        "value": "",
                      }
                    }
                    resource={
                      NamedNode {
                        "termType": "NamedNode",
                        "value": "bar",
                      }
                    }
                  />
                </a>
              </th>
              <td>
                baz
              </td>
            </tr>
          </tbody>,
          "responsiveTag": "div",
          "tag": "table",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <tr>
                <th
                  scope="row"
                >
                  <a
                    href="bar"
                  >
                    <Connect(Component)
                      graph={
                        DefaultGraph {
                          "termType": "DefaultGraph",
                          "value": "",
                        }
                      }
                      resource={
                        NamedNode {
                          "termType": "NamedNode",
                          "value": "bar",
                        }
                      }
                    />
                  </a>
                </th>
                <td>
                  baz
                </td>
              </tr>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "barbaz",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th
                    scope="row"
                  >
                    <a
                      href="bar"
                    >
                      <Connect(Component)
                        graph={
                          DefaultGraph {
                            "termType": "DefaultGraph",
                            "value": "",
                          }
                        }
                        resource={
                          NamedNode {
                            "termType": "NamedNode",
                            "value": "bar",
                          }
                        }
                      />
                    </a>
                  </th>,
                  <td>
                    baz
                  </td>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <a
                      href="bar"
                    >
                      <Connect(Component)
                        graph={
                          DefaultGraph {
                            "termType": "DefaultGraph",
                            "value": "",
                          }
                        }
                        resource={
                          NamedNode {
                            "termType": "NamedNode",
                            "value": "bar",
                          }
                        }
                      />
                    </a>,
                    "scope": "row",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Connect(Component)
                        graph={
                          DefaultGraph {
                            "termType": "DefaultGraph",
                            "value": "",
                          }
                        }
                        resource={
                          NamedNode {
                            "termType": "NamedNode",
                            "value": "bar",
                          }
                        }
                      />,
                      "href": "bar",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "graph": DefaultGraph {
                          "termType": "DefaultGraph",
                          "value": "",
                        },
                        "resource": NamedNode {
                          "termType": "NamedNode",
                          "value": "bar",
                        },
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": "a",
                  },
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "baz",
                  },
                  "ref": null,
                  "rendered": "baz",
                  "type": "td",
                },
              ],
              "type": "tr",
            },
          ],
          "type": "tbody",
        },
        "type": [Function],
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PropertyTable::render should render predicate-node pairs 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PropertyTable
    quads={
      Array [
        Quad {
          "graph": DefaultGraph {
            "termType": "DefaultGraph",
            "value": "",
          },
          "object": NamedNode {
            "termType": "NamedNode",
            "value": "baz",
          },
          "predicate": NamedNode {
            "termType": "NamedNode",
            "value": "bar",
          },
          "subject": NamedNode {
            "termType": "NamedNode",
            "value": "foo",
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Table
        responsiveTag="div"
        tag="table"
      >
        <tbody>
          <tr>
            <th
              scope="row"
            >
              <a
                href="bar"
              >
                <Connect(Component)
                  graph={
                    DefaultGraph {
                      "termType": "DefaultGraph",
                      "value": "",
                    }
                  }
                  resource={
                    NamedNode {
                      "termType": "NamedNode",
                      "value": "bar",
                    }
                  }
                />
              </a>
            </th>
            <td>
              <a
                href="baz"
              >
                <Connect(Component)
                  graph={
                    DefaultGraph {
                      "termType": "DefaultGraph",
                      "value": "",
                    }
                  }
                  resource={
                    NamedNode {
                      "termType": "NamedNode",
                      "value": "baz",
                    }
                  }
                />
              </a>
            </td>
          </tr>
        </tbody>
      </Table>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <tbody>
          <tr>
            <th
              scope="row"
            >
              <a
                href="bar"
              >
                <Connect(Component)
                  graph={
                    DefaultGraph {
                      "termType": "DefaultGraph",
                      "value": "",
                    }
                  }
                  resource={
                    NamedNode {
                      "termType": "NamedNode",
                      "value": "bar",
                    }
                  }
                />
              </a>
            </th>
            <td>
              <a
                href="baz"
              >
                <Connect(Component)
                  graph={
                    DefaultGraph {
                      "termType": "DefaultGraph",
                      "value": "",
                    }
                  }
                  resource={
                    NamedNode {
                      "termType": "NamedNode",
                      "value": "baz",
                    }
                  }
                />
              </a>
            </td>
          </tr>
        </tbody>,
        "responsiveTag": "div",
        "tag": "table",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <tr>
              <th
                scope="row"
              >
                <a
                  href="bar"
                >
                  <Connect(Component)
                    graph={
                      DefaultGraph {
                        "termType": "DefaultGraph",
                        "value": "",
                      }
                    }
                    resource={
                      NamedNode {
                        "termType": "NamedNode",
                        "value": "bar",
                      }
                    }
                  />
                </a>
              </th>
              <td>
                <a
                  href="baz"
                >
                  <Connect(Component)
                    graph={
                      DefaultGraph {
                        "termType": "DefaultGraph",
                        "value": "",
                      }
                    }
                    resource={
                      NamedNode {
                        "termType": "NamedNode",
                        "value": "baz",
                      }
                    }
                  />
                </a>
              </td>
            </tr>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "barbaz",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th
                  scope="row"
                >
                  <a
                    href="bar"
                  >
                    <Connect(Component)
                      graph={
                        DefaultGraph {
                          "termType": "DefaultGraph",
                          "value": "",
                        }
                      }
                      resource={
                        NamedNode {
                          "termType": "NamedNode",
                          "value": "bar",
                        }
                      }
                    />
                  </a>
                </th>,
                <td>
                  <a
                    href="baz"
                  >
                    <Connect(Component)
                      graph={
                        DefaultGraph {
                          "termType": "DefaultGraph",
                          "value": "",
                        }
                      }
                      resource={
                        NamedNode {
                          "termType": "NamedNode",
                          "value": "baz",
                        }
                      }
                    />
                  </a>
                </td>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    href="bar"
                  >
                    <Connect(Component)
                      graph={
                        DefaultGraph {
                          "termType": "DefaultGraph",
                          "value": "",
                        }
                      }
                      resource={
                        NamedNode {
                          "termType": "NamedNode",
                          "value": "bar",
                        }
                      }
                    />
                  </a>,
                  "scope": "row",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Connect(Component)
                      graph={
                        DefaultGraph {
                          "termType": "DefaultGraph",
                          "value": "",
                        }
                      }
                      resource={
                        NamedNode {
                          "termType": "NamedNode",
                          "value": "bar",
                        }
                      }
                    />,
                    "href": "bar",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "graph": DefaultGraph {
                        "termType": "DefaultGraph",
                        "value": "",
                      },
                      "resource": NamedNode {
                        "termType": "NamedNode",
                        "value": "bar",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "a",
                },
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    href="baz"
                  >
                    <Connect(Component)
                      graph={
                        DefaultGraph {
                          "termType": "DefaultGraph",
                          "value": "",
                        }
                      }
                      resource={
                        NamedNode {
                          "termType": "NamedNode",
                          "value": "baz",
                        }
                      }
                    />
                  </a>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Connect(Component)
                      graph={
                        DefaultGraph {
                          "termType": "DefaultGraph",
                          "value": "",
                        }
                      }
                      resource={
                        NamedNode {
                          "termType": "NamedNode",
                          "value": "baz",
                        }
                      }
                    />,
                    "href": "baz",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "graph": DefaultGraph {
                        "termType": "DefaultGraph",
                        "value": "",
                      },
                      "resource": NamedNode {
                        "termType": "NamedNode",
                        "value": "baz",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "a",
                },
                "type": "td",
              },
            ],
            "type": "tr",
          },
        ],
        "type": "tbody",
      },
      "type": [Function],
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Table
          responsiveTag="div"
          tag="table"
        >
          <tbody>
            <tr>
              <th
                scope="row"
              >
                <a
                  href="bar"
                >
                  <Connect(Component)
                    graph={
                      DefaultGraph {
                        "termType": "DefaultGraph",
                        "value": "",
                      }
                    }
                    resource={
                      NamedNode {
                        "termType": "NamedNode",
                        "value": "bar",
                      }
                    }
                  />
                </a>
              </th>
              <td>
                <a
                  href="baz"
                >
                  <Connect(Component)
                    graph={
                      DefaultGraph {
                        "termType": "DefaultGraph",
                        "value": "",
                      }
                    }
                    resource={
                      NamedNode {
                        "termType": "NamedNode",
                        "value": "baz",
                      }
                    }
                  />
                </a>
              </td>
            </tr>
          </tbody>
        </Table>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <tbody>
            <tr>
              <th
                scope="row"
              >
                <a
                  href="bar"
                >
                  <Connect(Component)
                    graph={
                      DefaultGraph {
                        "termType": "DefaultGraph",
                        "value": "",
                      }
                    }
                    resource={
                      NamedNode {
                        "termType": "NamedNode",
                        "value": "bar",
                      }
                    }
                  />
                </a>
              </th>
              <td>
                <a
                  href="baz"
                >
                  <Connect(Component)
                    graph={
                      DefaultGraph {
                        "termType": "DefaultGraph",
                        "value": "",
                      }
                    }
                    resource={
                      NamedNode {
                        "termType": "NamedNode",
                        "value": "baz",
                      }
                    }
                  />
                </a>
              </td>
            </tr>
          </tbody>,
          "responsiveTag": "div",
          "tag": "table",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <tr>
                <th
                  scope="row"
                >
                  <a
                    href="bar"
                  >
                    <Connect(Component)
                      graph={
                        DefaultGraph {
                          "termType": "DefaultGraph",
                          "value": "",
                        }
                      }
                      resource={
                        NamedNode {
                          "termType": "NamedNode",
                          "value": "bar",
                        }
                      }
                    />
                  </a>
                </th>
                <td>
                  <a
                    href="baz"
                  >
                    <Connect(Component)
                      graph={
                        DefaultGraph {
                          "termType": "DefaultGraph",
                          "value": "",
                        }
                      }
                      resource={
                        NamedNode {
                          "termType": "NamedNode",
                          "value": "baz",
                        }
                      }
                    />
                  </a>
                </td>
              </tr>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "barbaz",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th
                    scope="row"
                  >
                    <a
                      href="bar"
                    >
                      <Connect(Component)
                        graph={
                          DefaultGraph {
                            "termType": "DefaultGraph",
                            "value": "",
                          }
                        }
                        resource={
                          NamedNode {
                            "termType": "NamedNode",
                            "value": "bar",
                          }
                        }
                      />
                    </a>
                  </th>,
                  <td>
                    <a
                      href="baz"
                    >
                      <Connect(Component)
                        graph={
                          DefaultGraph {
                            "termType": "DefaultGraph",
                            "value": "",
                          }
                        }
                        resource={
                          NamedNode {
                            "termType": "NamedNode",
                            "value": "baz",
                          }
                        }
                      />
                    </a>
                  </td>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <a
                      href="bar"
                    >
                      <Connect(Component)
                        graph={
                          DefaultGraph {
                            "termType": "DefaultGraph",
                            "value": "",
                          }
                        }
                        resource={
                          NamedNode {
                            "termType": "NamedNode",
                            "value": "bar",
                          }
                        }
                      />
                    </a>,
                    "scope": "row",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Connect(Component)
                        graph={
                          DefaultGraph {
                            "termType": "DefaultGraph",
                            "value": "",
                          }
                        }
                        resource={
                          NamedNode {
                            "termType": "NamedNode",
                            "value": "bar",
                          }
                        }
                      />,
                      "href": "bar",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "graph": DefaultGraph {
                          "termType": "DefaultGraph",
                          "value": "",
                        },
                        "resource": NamedNode {
                          "termType": "NamedNode",
                          "value": "bar",
                        },
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": "a",
                  },
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <a
                      href="baz"
                    >
                      <Connect(Component)
                        graph={
                          DefaultGraph {
                            "termType": "DefaultGraph",
                            "value": "",
                          }
                        }
                        resource={
                          NamedNode {
                            "termType": "NamedNode",
                            "value": "baz",
                          }
                        }
                      />
                    </a>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Connect(Component)
                        graph={
                          DefaultGraph {
                            "termType": "DefaultGraph",
                            "value": "",
                          }
                        }
                        resource={
                          NamedNode {
                            "termType": "NamedNode",
                            "value": "baz",
                          }
                        }
                      />,
                      "href": "baz",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "graph": DefaultGraph {
                          "termType": "DefaultGraph",
                          "value": "",
                        },
                        "resource": NamedNode {
                          "termType": "NamedNode",
                          "value": "baz",
                        },
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": "a",
                  },
                  "type": "td",
                },
              ],
              "type": "tr",
            },
          ],
          "type": "tbody",
        },
        "type": [Function],
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
